include/ipz_parser.hpp

```diff
@@ -28,20 +28,22 @@ class IpzVpdParser : public ParserInterface
     /**
      * @brief Constructor.
      *
-     * @param[in] vpdVector - VPD data.
+     * @param[in, out] vpdVector - VPD data.
      * @param[in] vpdFilePath - Path to VPD EEPROM.
      * @param[in] vpdStartOffset - Offset from where VPD starts in the file.
      * Defaulted to 0.
      */
-    IpzVpdParser(const types::BinaryVector& vpdVector,
-                 const std::string& vpdFilePath, size_t vpdStartOffset = 0) :
+    IpzVpdParser(types::BinaryVector vpdVector, const std::string& vpdFilePath,
```

Comment:
Why this needs to change? Am I missing something here?
Also, this is an input variable needed to parse the data. Removing constness sacrifices the control on keeping the data clean. Will need to see, only if there is no other way.

Reply:
The constructor cater to the complete class. There are other functionalities as well being offered by the class.
Making it non const is a localized requirement of write operation and hence should be handled locally.